loadflow/loadflow-api/src/main/java/com/powsybl/loadflow/json/LoadFlowResultDeserializer.java

```diff
@@ -34,17 +34,25 @@
     }
 
     public LoadFlowResult.ComponentResult deserializeComponentResult(JsonParser parser) throws IOException {
-        Integer componentNum = null;
+        Integer connectedComponentNum = null;
+        Integer synchronousComponentNum = null;
         LoadFlowResult.ComponentResult.Status status = null;
         Integer iterationCount = null;
         String slackBusId = null;
         Double slackBusActivePowerMismatch = null;
 
         while (parser.nextToken() != JsonToken.END_OBJECT) {
             switch (parser.getCurrentName()) {
+                case "connectedComponentNum":
+                    parser.nextToken();
+                    connectedComponentNum = parser.getValueAsInt();
+                    break;
+
+                case "synchronousComponentNum":
+
                 case "componentNum":
```

Comment:
Doing so means that we can read a malformed json file containing a result with componentNum and connectedComponentNum instead of throwing an error. I think you should test here which version you're reading.

Reply:
Yes. I just added some exceptions throwing when loading unexpected fields with a specific version.